### ISSUE TITLE
redirecttohttps bug

### ISSUE THREAD

**Issue by binbooly** [gotify/server#519]
server:
  ...
  ssl:
    enabled: true
    redirecttohttps: true
    port: 8443

Should redirect to "https://www.abc.com:8443/".

**Comment by jmattheis** [comment]
Cannot reproduce.
```
$ cat config.yml
server:
  ssl:
    enabled: true
    redirecttohttps:
    port: 8443 # the https port
    certfile: 'server.crt'
    certkey: 'server.key'
  cors:
    alloworigins:
      - ".*"
  stream:
    allowedorigins:
     - ".*"
  port: 8080 # the port the HTTP server will listen on
$ ./gotify-linux-amd64
Starting Gotify version 2.1.7@2022-09-11-07:29:06
Started Listening for TLS connection on :8443
Started Listening for plain HTTP connection on :8080

$ curl -vv localhost:8080
*   Trying 127.0.0.1:8080...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET / HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/7.86.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 302 Found
< Content-Type: text/html; charset=utf-8
< Location: https://localhost:8443/
< Date: Tue, 08 Nov 2022 17:58:50 GMT
< Content-Length: 46
<
<a href="https://localhost:8443/">Found</a>.

```

**Comment by binbooly** [comment]
> Cannot reproduce.
> 
> ```
> $ cat config.yml
> server:
>   ssl:
>     enabled: true
>     redirecttohttps:
>     port: 8443 # the https port
>     certfile: 'server.crt'
>     certkey: 'server.key'
>   cors:
>     alloworigins:
>       - ".*"
>   stream:
>     allowedorigins:
>      - ".*"
>   port: 8080 # the port the HTTP server will listen on
> $ ./gotify-linux-amd64
> Starting Gotify version 2.1.7@2022-09-11-07:29:06
> Started Listening for TLS connection on :8443
> Started Listening for plain HTTP connection on :8080
> 
> $ curl -vv localhost:8080
> *   Trying 127.0.0.1:8080...
> * Connected to localhost (127.0.0.1) port 8080 (#0)
> > GET / HTTP/1.1
> > Host: localhost:8080
> > User-Agent: curl/7.86.0
> > Accept: */*
> >
> * Mark bundle as not supporting multiuse
> < HTTP/1.1 302 Found
> < Content-Type: text/html; charset=utf-8
> < Location: https://localhost:8443/
> < Date: Tue, 08 Nov 2022 17:58:50 GMT
> < Content-Length: 46
> <
> <a href="https://localhost:8443/">Found</a>.
> ```

```
server:
  ssl:
    enabled: true
    redirecttohttps: true
    port: 8443 # the https port
    certfile: server.pem
    certkey: server.key
  cors:
    alloworigins:
      - ".*"
  stream:
    allowedorigins:
     - ".*"
  port: 80


[root@localhost gotify]# curl -vv http://18.183.147.9
* About to connect() to 18.183.147.9 port 80 (#0)
*   Trying 18.183.147.9...
* Connected to 18.183.147.9 (18.183.147.9) port 80 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.29.0
> Host: 18.183.147.9
> Accept: */*
> 
< HTTP/1.1 302 Found
< Content-Type: text/html; charset=utf-8
< Location: https://18.183.147.9/
< Date: Wed, 30 Nov 2022 18:49:17 GMT
< Content-Length: 44
< 
<a href="https://18.183.147.9/">Found</a>.

* Connection #0 to host 18.183.147.9 left intact
```

**Comment by jmattheis** [comment]
Fixed with [Version 2.2.0](https://github.com/gotify/server/releases/tag/v2.2.0)
